### PR TITLE
Immovable block support

### PR DESCRIPTION
**Pull request by geppy**
Should there be indications that immovable blocks are immovable?

I put the annotation in a separate commit in case you don't want to merge it (I'm not sure if it's idiomatic or not).

---
This change is [Reviewable](https://reviewable.io/reviews/google/blockly-android/332)

### CODE REVIEW

**Comment by geppy** [comment]
This PR is intended to address #304.  Are there any other avenues that need to be addressed?  That is, can end users move blocks via non-dragging methods (e.g. via a11y tools)?

File: blocklylib-core/src/main/java/com/google/blockly/android/ui/BlockView.java [hunk 1]
```diff
@@ -37,6 +38,7 @@
     /**
      * @return The block represented by this view.
      */
+    @NonNull
```
Comment (AnmAtAnm) [inline blocklylib-core/src/main/java/com/google/blockly/android/ui/BlockView.java]: Worth adding to AbstractBlockView, too.

**Comment by AnmAtAnm** [comment]
Generally looks good.  I'm curious if you've tried this out.  Does the event bubble up to allow the workspace to pan?  Also, can we get an new test case in DraggerTest for this?

**Comment by geppy** [comment]
I've tested this manually, but I'm working on creating Espresso-based unit tests.  I'll create a test for panning.

**Comment by AnmAtAnm** [comment]
Thanks, @geppy. You rock!

**Comment by RoboErikG** [comment]
It looks like this would let the event bubble up and be handled by the Workspace, though that's the current behavior on the web as well so seems reasonable.

Please add a test, but otherwise lgtm. Thanks, geppy!

**Comment by AnmAtAnm** [comment]
> Should there be indications that immovable blocks are immovable?

Non-movable blocks do not have any distinguishing rendering compared to movable block. Thanks for asking.

**Comment by AnmAtAnm** [comment]
@geppy
Just checking in.  Have you found any time to work on this, or run into any problems along the way?

**Comment by geppy** [comment]
@AnmAtAnm Thanks for the reminder, I got sidetracked by another project.  I'll try to get this wrapped up this weekend.

**Pull request merged**